internal/expr/expr_test.go

```diff
@@ -655,6 +668,70 @@ AND z = @sqlair_0 -- The line with $Person.id on it
+}, {
+	summary:        "bulk insert with explicit columns and constants",
+	query:          `INSERT INTO person (name, mapKey, street) VALUES ($Person.*, $Address.*, $M.*)`,
+	expectedParsed: `[Bypass[INSERT INTO person ] ColumnInsert[[name mapKey street] [Person.* Address.* M.*]]]`,
+	typeSamples:    []any{Person{}, Address{}, M{}},
+	inputArgs:      []any{[]Person{{Fullname: "Al"}, {Fullname: "Albert"}}, M{"mapKey": "const map"}, Address{Street: "const street"}},
+	expectedParams: []any{"Al", "const map", "const street", "Albert", "const map", "const street"},
+	expectedSQL:    `INSERT INTO person (name, mapKey, street) VALUES (@sqlair_0, @sqlair_1, @sqlair_2), (@sqlair_3, @sqlair_4, @sqlair_5)`,
+}, {
+	summary:        "bulk insert with omit empty struct not inserted at all",
+	query:          `INSERT INTO person (*) VALUES ($OmitEmptyID.*, $M.key)`,
+	expectedParsed: `[Bypass[INSERT INTO person ] AsteriskInsert[[*] [OmitEmptyID.* M.key]]]`,
+	typeSamples:    []any{OmitEmptyID{}, M{}},
+	inputArgs:      []any{[]OmitEmptyID{{ID: 0}, {ID: 0}}, M{"key": "val"}},
+	expectedParams: []any{"val", "val"},
+	expectedSQL:    `INSERT INTO person (key) VALUES (@sqlair_0), (@sqlair_1)`,
+}, {
+	summary:        "bulk insert with multiple bulk inputs",
+	query:          `INSERT INTO person (*) VALUES ($Person.id, $M.key, $Address.street)`,
+	expectedParsed: `[Bypass[INSERT INTO person ] AsteriskInsert[[*] [Person.id M.key Address.street]]]`,
+	typeSamples:    []any{Person{}, M{}, Address{}},
+	inputArgs:      []any{[]Person{{ID: 1}, {ID: 2}}, M{"key": "val"}, []Address{{Street: "Main Street"}, {Street: "Church Road"}}},
+	expectedParams: []any{1, "val", "Main Street", 2, "val", "Church Road"},
+	expectedSQL:    `INSERT INTO person (id, key, street) VALUES (@sqlair_0, @sqlair_1, @sqlair_2), (@sqlair_3, @sqlair_4, @sqlair_5)`,
+}, {
+	summary:        "bulk insert with pointers",
+	query:          `INSERT INTO person (*) VALUES ($Person.id, $M.key)`,
+	expectedParsed: `[Bypass[INSERT INTO person ] AsteriskInsert[[*] [Person.id M.key]]]`,
+	typeSamples:    []any{Person{}, M{}},
+	inputArgs:      []any{[]*Person{{ID: 1}, {ID: 2}}, []*M{{"key": "val1"}, {"key": "val2"}}},
+	expectedParams: []any{1, "val1", 2, "val2"},
+	expectedSQL:    `INSERT INTO person (id, key) VALUES (@sqlair_0, @sqlair_1), (@sqlair_2, @sqlair_3)`,
+}, {
+	summary:        "bulk insert with slices of length 1",
+	query:          `INSERT INTO person (*) VALUES ($Person.id, $M.key, $Address.street)`,
+	expectedParsed: `[Bypass[INSERT INTO person ] AsteriskInsert[[*] [Person.id M.key Address.street]]]`,
+	typeSamples:    []any{Person{}, M{}, Address{}},
+	inputArgs:      []any{[]Person{{ID: 1}}, M{"key": "val"}, []Address{{Street: "Church Road"}}},
+	expectedParams: []any{1, "val", "Church Road"},
+	expectedSQL:    `INSERT INTO person (id, key, street) VALUES (@sqlair_0, @sqlair_1, @sqlair_2)`,
+}, {
+	summary:        "bulk insert with explicit columns and literal",
```

Comment:
This test is the same as the one below, right?

Reply:
Yea pretty much, the second is a more complex case of the first, though I'm inclined to keep both, because these also serve as examples to people looking at the library. 

I've removed the third insert value from the first one so that it at least tests a literal at the end.